src/main/java/Duke.java

```diff
@@ -1,10 +1,195 @@
+            if (temp.startsWith("[T]")){
+                newTask = new Todo(temp.substring(7));
+                t.add(newTask);
+            }
+
+            else if (temp.startsWith("[E]")){
+                newTask = new Event(temp.substring(7, temp.indexOf(" (")),
+                        temp.substring(temp.indexOf("(at:") + 5, temp.indexOf(")")));
+                t.add(newTask);
+            }
+
+            else if (temp.startsWith("[D]")){
+                newTask = new Deadline(temp.substring(7, temp.indexOf(" (")),
+                        temp.substring(temp.indexOf("(by:") + 5, temp.indexOf(")")));
+                t.add(newTask);
+            }
+
+            index++;
+        }
+        return index;
+    }
+
+
+    public static int addTask(ArrayList<Task> t, String message, int index) throws DukeException {
+
+        // list
+        if (message.equals("list")){
+            for (int i = 0; i < index; i++) {
+                System.out.print(i + 1 + ". ");
+                System.out.println(t.get(i));
+            }
+        }
+
+        // mark as done
+        else if (message.startsWith("done")){
+            int l = Integer.parseInt(message.substring(5)) - 1;
+            t.get(l).markAsDone();
+        }
+
+        // new deadline
+        else if (message.startsWith("deadline")){
+            t.add(new Deadline(message.substring(9, message.indexOf(" /")),
+                    message.substring(message.indexOf("/by") + 4)));
+            acknowledgement(t.get(index), "add",index+1);
+            index++;
+        }
+
+        // new todo
+        else if (message.startsWith("todo")){
+            t.add(new Todo(message.substring(5)));
+            acknowledgement(t.get(index), "add", index+1);
+            index++;
+        }
+
+        // new event
+        else if (message.startsWith("event")){
+            t.add(new Event(message.substring(6, message.indexOf(" /")),
+                    message.substring(message.indexOf("/at") + 4)));
+            acknowledgement(t.get(index), "add", index+1);
+            index++;
+        }
+
+        // delete task
+        else if (message.startsWith("delete")){
+            int l = Integer.parseInt(message.substring(7)) - 1;
+            acknowledgement(t.get(l), "del", --index);
+            t.remove(t.get(l));
+        }
+
+        // throw exception
+        else {
+            throw new DukeException();
+        }
+
+        prepareWriteToFile(t);
+        return index;
+    }
+
+    public static void acknowledgement(Task message, String note, int i){
+        if (note.equals("add")) {
+            System.out.println("Got it. I've added this task:");
+        }
+        else {
+            System.out.println("Got it. I've removed this task:");
+        }
+        System.out.println("    " + message);
+        System.out.println("Now you have " + i + (i == 1 ? " task" : " tasks") + " in your list.");
+    }
+
+    public static void prepareWriteToFile (ArrayList<Task> tasks){
```

Comment:
Consider returning the string from this method and then pass that string to the next method to _actually_ write to the file.
When you do that, rename this method appropriately